plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -0,0 +1,389 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.hdfs.HdfsFileSystemFactory;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorSession;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.plugin.iceberg.IcebergUtil.METADATA_FOLDER_NAME;
+import static io.trino.plugin.iceberg.util.IcebergRegisterTableUtil.getLatestMetadataLocation;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestIcebergRegisterTableProcedure
+        extends AbstractTestQueryFramework
+{
+    private HiveMetastore metastore;
+    private File metastoreDir;
+    private TrinoFileSystem trinoFileSystem;
+    private static final String TEST_SCHEMA_NAME = "test_schema";
+    private static final String CATALOG = "iceberg";
+
+    public enum StorageFormat
+    {
+        PARQUET("parquet"),
+        ORC("orc"),
+        AVRO("avro"),
+        /**/;
+
+        private final String name;
+
+        StorageFormat(String format)
+        {
+            this.name = format;
+        }
+
+        @Override
+        public String toString()
+        {
+            return this.name;
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        metastoreDir = Files.createTempDirectory("test_iceberg_register_table").toFile();
+        metastore = createTestingFileHiveMetastore(metastoreDir);
+        trinoFileSystem = new HdfsFileSystemFactory(HDFS_ENVIRONMENT).create(TestingConnectorSession.SESSION);
+        return IcebergQueryRunner.builder()
+                .setMetastoreDirectory(metastoreDir)
+                .build();
+    }
+
+    @BeforeClass
+    @Override
+    public void init()
+            throws Exception
+    {
+        super.init();
```

Comment:
don't overwrite init methods

have your own

Reply:
I was trying to create a new schema. So overridden. Will remove it as we will be using `CURRET_SCHEMA`

Reply:
Removed it. using `CURRENT_SCHEMA` now